src/VisualStudio/Core/Def/Implementation/ProjectSystem/MetadataReferences/FileWatchedPortableExecutableReferenceFactory.cs

```diff
@@ -53,9 +54,15 @@ internal sealed class FileWatchedPortableExecutableReferenceFactory
         {
             _visualStudioWorkspace = visualStudioWorkspace;
 
-            // TODO: set this to watch the NuGet directory or the reference assemblies directory; since those change rarely and most references
-            // will come from them, we can avoid creating a bunch of explicit file watchers.
-            _fileReferenceChangeContext = fileChangeWatcherProvider.Watcher.CreateContext();
+            // We will do a single directory watch on the Reference Assemblies folder to avoid having to create separate file
+            // watches on individual .dlls that effectively never change.
+            string referenceAssembliesPath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ProgramFilesX86), "Reference Assemblies", "Microsoft", "Framework");
+            var referenceAssemblies = new FileChangeWatcher.WatchedDirectory(referenceAssembliesPath, ".dll");
```

Comment:
is it gauranteed this folder exists?  if it doesn't do we throw?  or just silently go on?

Reply:
The only way it wouldn't exist is you don't have any .NET Framework SDK at all on the machine. If it didn't exist MiscellaneousFilesWorkspace would also be broken, because there'd be no references to pick up at all either for that.